nifi-api/src/main/java/org/apache/nifi/controller/status/ConnectionStatus.java

```diff
@@ -22,6 +22,7 @@
 /**
  */
 public class ConnectionStatus implements Cloneable {
+    private long createdAtInMs;
```

Comment:
This addition raises several questions and concerns. It does not seem like the NiFi API definition should include the creation timestamp of components in all circumstances. It would make much more sense to track this as a feature internal to the framework.

As far as naming the `InMs` suffix is awkward, and it would be much better to use `java.time.Instant` for model classes.

Reply:
Based on most usage I would think a time related information would make sense within status (at what time the status snapshot is relevant). I am not sure this is true for all usages but the current setup on main (these two coupled information travels together) does not seem straightforward to me. I cannot rule out however a possible third solution which complies with both things.

Instant is a good idea. I think I went down this road because some already contained creation time in this flavor but the `Instant` really makes more sense (especially that we convert it to instant many places already)

Reply:
Although it might require some additional replication in the QuestDB implementation itself, it is important to avoid introducing specific concepts to the NiFi API that do not have general applicability. I would have to take a closer look at how this is used, but in general, I believe any changes to the nifi-api should be avoided for the purpose of this implementation.

Reply:
The implementation was the trigger for the change but not the sole reason. The idea was -based on how we use these- that the state snapshot and the time of relevance travels together in many times which makes it suspicious that they should be boundled together. I will not touch until you make your rounds, let's discuss after that how you feel about the approach.